otx/algorithms/classification/tasks/inference.py

```diff
@@ -319,14 +321,19 @@ def _init_recipe(self):
     # error log : ValueError: Unexpected type of 'data_loader' parameter
     def _init_model_cfg(self):
         base_dir = os.path.abspath(os.path.dirname(self.template_file_path))
+
+        model_base_name = "model"
+        if self._hyperparams.learning_parameters.enable_supcon:
+            model_base_name = "model_supcon"
+
         if self._multilabel:
             cfg_path = os.path.join(base_dir, "model_multilabel.py")
         elif self._hierarchical:
             cfg_path = os.path.join(base_dir, "model_hierarchical.py")
         else:
-            cfg_path = os.path.join(base_dir, "model.py")
-        cfg = MPAConfig.fromfile(cfg_path)
+            cfg_path = os.path.join(base_dir, model_base_name + ".py")
```

Comment:
How about using one additional `elif` sentence for `supcon` to unify the codes?

AS IS
```
if self._hyperparams.learning_parameters.enable_supcon:
            model_base_name = "model_supcon"

        if self._multilabel:
            cfg_path = os.path.join(base_dir, "model_multilabel.py")
        elif self._hierarchical:
            cfg_path = os.path.join(base_dir, "model_hierarchical.py")
        else:
            cfg_path = os.path.join(base_dir, model_base_name + ".py")
``` 

TO BE
```
        if self._multilabel:
            cfg_path = os.path.join(base_dir, "model_multilabel.py")
        elif self._hierarchical:
            cfg_path = os.path.join(base_dir, "model_hierarchical.py")
        elif self._hyperparameters.learning_parameters.enable_supcon:
            cfg_path = os.path.join(base_dir, "model_supcon.py")
        else:
            cfg_path = os.path.join(base_dir, "model.py")
```

Reply:
And actually, I want to abbreviate `self._hyperparameters.learning_parameters.enable_supcon` --> `self._supcon` by adding some codes.

Reply:
I wrote it like this because the `enable_supcon` toggle is not at the same level (or exclusive) with the other `TrainType`s, but complimentary. So, it seems misleading to me.

Moreover, in an upcoming PR, there will be `model_supcon_multilabel` and `model_supcon_hierarchical`.

Reply:
> I wrote it like this because the `enable_supcon` toggle is not at the same level (or exclusive) with the other `TrainType`s, but complimentary. So, it seems misleading to me.
> 
> Moreover, in an upcoming PR, there will be `model_supcon_multilabel` and `model_supcon_hierarchical`.

If so, the name of `model_supcon.py` is weird to me because the goal of `model_****.py`  is to divide the `TrainType`.
Similarly, `model_supcon_multilabel` and `model_supcon_hierarchical` are also weird. As you said, although `supcon` represents the complementary method and `multilabel` represents the `TrainType`, they'll be located the same level.

Reply:
However, actually, in my case, I don't have ideas for the better way, now.